posthog/management/commands/backfill_sessions_table.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import Optional
+
+import structlog
+from django.core.management.base import BaseCommand, CommandError
+
+from posthog.clickhouse.client.execute import sync_execute
+from posthog.models.property.util import get_property_string_expr
+from posthog.models.team.team import Team
+
+logger = structlog.get_logger(__name__)
+
+
+TARGET_TABLE = "sessions"
+
+
+@dataclass
+class BackfillQuery:
+    team_id: int
+
+    def execute(self, dry_run: bool = True, month: Optional[str] = None) -> None:
+        where = "true"
+        if month is not None:
+            where = f"toYYYYMM(timestamp) < {month}"
+
+        def source_column(column_name: str) -> str:
+            return get_property_string_expr(
+                "events", property_name=column_name, var=f"'{column_name}'", column="properties"
+            )[0]
+
+        current_url_property = source_column("$current_url")
+        referring_domain_property = source_column("$referring_domain")
+        utm_source_property = source_column("utm_source")
+        utm_campaign_property = source_column("utm_campaign")
+        utm_medium_property = source_column("utm_medium")
+        utm_term_property = source_column("utm_term")
+        utm_content_property = source_column("utm_content")
+        gclid_property = source_column("gclid")
+        gad_source_property = source_column("gad_source")
+
+        select_query = f"""
+SELECT
+    `$session_id` as session_id,
+    team_id,
+
+    distinct_id,
+
+    timestamp AS min_first_timestamp,
+    timestamp AS max_last_timestamp,
+
+    [{current_url_property}] AS urls,
+    initializeAggregation('argMinState', {current_url_property}, timestamp) as entry_url,
+    initializeAggregation('argMaxState', {current_url_property}, timestamp) as exit_url,
+
+    initializeAggregation('argMinState', {referring_domain_property}, timestamp) as initial_referring_domain,
+    initializeAggregation('argMinState', {utm_source_property}, timestamp) as initial_utm_source,
+    initializeAggregation('argMinState', {utm_campaign_property}, timestamp) as initial_utm_campaign,
+    initializeAggregation('argMinState', {utm_medium_property}, timestamp) as initial_utm_medium,
+    initializeAggregation('argMinState', {utm_term_property}, timestamp) as initial_utm_term,
+    initializeAggregation('argMinState', {utm_content_property}, timestamp) as initial_utm_content,
+    initializeAggregation('argMinState', {gclid_property}, timestamp) as initial_gclid,
+    initializeAggregation('argMinState', {gad_source_property}, timestamp) as initial_gad_source,
+
+    CAST(([event], [1]), 'Map(String, UInt64)') as event_count_map,
+    if(event='$pageview', 1, NULL) as pageview_count,
+    if(event='$autocapture', 1, NULL) as autocapture_count
+
+FROM events
+WHERE team_id = %(team_id)s AND `$session_id` IS NOT NULL AND `$session_id` != '' AND {where}
```

Comment:
This is a great django mgmt command to backfill, nice job.

I do see a risk of the query here diverging from what is defined in the materialized view - might be worth drawing from the same place, but that's not a blocker here